azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1323,6 +1327,8 @@ private void addInitContainers(final int executionId,
       final AzKubernetesV1SpecBuilder v1SpecBuilder,
       final VersionSet versionSet)
       throws ExecutorManagerException {
+    final ExecutableFlow flow = this.executorLoader.fetchExecutableFlow(executionId);
```

Comment:
I would suggest pass execFlow object from createPodSpec() to addInitContainers() to save one DB fetch.